### PR TITLE
Open source guidelines

### PR DESCRIPTION
**Pull request by codevelvet**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+[fork]: https://help.github.com/articles/fork-a-repo/
+[branch]: https://help.github.com/articles/creating-and-deleting-branches-within-your-repository/
+[pr]: https://help.github.com/articles/using-pull-requests/
+
+## General Guidelines
+
+* When in doubt, test it.  If you can't test it, re-think what you are doing.
+* Code formatting and internal application architecture should be consistent.
+* If you are changing UI, include screenshots or screencasts of the change.
```
Comment (davetron5000) [inline CONTRIBUTING.md]: Does this apply to code libraries?

Comment (codevelvet) [inline CONTRIBUTING.md]: UI? No, I can remove.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Contributing
```
Comment (simeonwillbanks) [inline CONTRIBUTING.md]: Before listing the steps, why don't we setup the process?

Comment (simeonwillbanks) [inline CONTRIBUTING.md]: Thanks for using and improving *ImmutableStruct*! If you'd like to help out, check out [the project's issues list][issues] for ideas on what could be improved.  If there's an idea you'd like to propose, or a design change, feel free to
file a new issue or send a pull request:

1. [Fork][fork] the repo.

...

[issues]: https://github.com/stitchfix/immutable-struct/issues
[fork]: https://help.github.com/articles/fork-a-repo/

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+1. [Fork][fork] the repo.
+2. [Create a topic branch.][branch]
+3. Write tests.
+4. Implement your feature or fix bug.
+5. Add, commit, and push your changes.
+6. [Submit a pull request.][pr]
```
Comment (simeonwillbanks) [inline CONTRIBUTING.md]: These can be an ordered list, so editing is easier. 

```
 1. [Fork][fork] the repo.
 1. [Create a topic branch.][branch]
 1. Write tests.
 1. Implement your feature or fix bug.
 1. Add, commit, and push your changes.
 1. [Submit a pull request.][pr]
```

File: CONTRIBUTING.md (hunk 2 above)
Comment (codevelvet) [inline CONTRIBUTING.md]: Yes, that sounds good.

Comment (davetron5000) [inline CONTRIBUTING.md]: :+1:

**Comment by codevelvet** [comment]
Added the code of conduct.

**Comment by davetron5000** [comment]
:+1:

**Pull request merged**